### PR TITLE
Refresh plugin for October 2023

### PR DESCRIPTION
**Pull request by basil**
Update build and test harness to latest toolchain versions. See self-review for explanation of changes.

### CODE REVIEW

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
```
Comment (basil) [inline .mvn/extensions.xml:1]: See the latest archetype and https://github.com/jenkinsci/incrementals-tools

File: .mvn/maven.config [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-Pconsume-incrementals
```
Comment (basil) [inline .mvn/maven.config:1]: Ditto

File: Jenkinsfile [hunk 3]
```diff
@@ -1 +1,4 @@
-buildPlugin()
+buildPlugin(useContainerAgent: true, configurations: [
```
Comment (basil) [inline Jenkinsfile:1]: As per the latest archetype

File: pom.xml [hunk 4]
```diff
@@ -1,13 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (basil) [inline pom.xml:1]: As per archetype

File: pom.xml [hunk 5]
```diff
@@ -1,13 +1,15 @@
...
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>4.56</version>
+    <version>4.75</version>
```
Comment (basil) [inline pom.xml:7]: Latest plugin parent POM

File: pom.xml [hunk 6]
```diff
@@ -1,13 +1,15 @@
...
+    <relativePath />
   </parent>
 
   <artifactId>aws-codepipeline</artifactId>
-  <version>0.47-SNAPSHOT</version>
+  <version>${revision}${changelist}</version>
```
Comment (basil) [inline pom.xml:12]: Incrementals

File: pom.xml [hunk 7]
```diff
@@ -17,7 +19,7 @@
     and the post-build event uploads the build artifacts and calls
     PutJobSuccessResult/PutJobFailureResult to report back the build status.
   </description>
-  <url>https://github.com/jenkinsci/aws-codepipeline-plugin</url>
+  <url>https://github.com/jenkinsci/${project.artifactId}-plugin</url>
```
Comment (basil) [inline pom.xml:22]: See archetype

File: pom.xml [hunk 8]
```diff
@@ -36,25 +38,25 @@
...
 
   <scm>
-    <url>https://github.com/jenkinsci/aws-codepipeline-plugin</url>
-    <connection>scm:git:ssh://github.com/jenkinsci/aws-codepipeline-plugin.git</connection>
-    <developerConnection>scm:git:ssh://git@github.com/jenkinsci/aws-codepipeline-plugin.git</developerConnection>
-    <tag>HEAD</tag>
+    <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
+    <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
+    <url>https://github.com/${gitHubRepo}</url>
+    <tag>${scmTag}</tag>
```
Comment (basil) [inline pom.xml:44]: Archetype/incrementals

File: pom.xml [hunk 9]
```diff
@@ -36,25 +38,25 @@
...
   <repositories>
     <repository>
-        <id>repo.jenkins-ci.org</id>
-        <url>https://repo.jenkins-ci.org/public/</url>
-    </repository>
-    <repository>
-        <id>lib</id>
-        <url>file://${project.basedir}/lib</url>
+      <id>repo.jenkins-ci.org</id>
+      <url>https://repo.jenkins-ci.org/public/</url>
```
Comment (basil) [inline pom.xml:50]: Archetype

File: pom.xml [hunk 10]
```diff
@@ -69,26 +71,14 @@
       <artifactId>aws-java-sdk-s3</artifactId>
       <version>${aws-sdk-version}</version>
     </dependency>
-    <dependency>
-      <groupId>org.apache.commons</groupId>
-      <artifactId>commons-compress</artifactId>
-      <version>1.24.0</version>
-    </dependency>
```
Comment (basil) [inline pom.xml:76]: Provided by Jenkins core, so no need to bundle our own copy.

File: pom.xml [hunk 11]
```diff
@@ -69,26 +71,14 @@
...
       <groupId>joda-time</groupId>
       <artifactId>joda-time</artifactId>
       <version>2.9.9</version>
     </dependency>
-    <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <version>4.13.2</version>
-      <scope>test</scope>
-    </dependency>
```
Comment (basil) [inline pom.xml:87]: Managed in plugin parent POM.

File: pom.xml [hunk 12]
```diff
@@ -69,26 +71,14 @@
...
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
-      <version>1.10.19</version>
```
Comment (basil) [inline pom.xml:91]: Managed in plugin parent POM.

File: pom.xml [hunk 13]
```diff
@@ -99,31 +89,4 @@
...
-        </configuration>
-      </plugin>
-
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-release-plugin</artifactId>
-        <version>2.5.2</version>
-      </plugin>
-    </plugins>
-  </build>
```
Comment (basil) [inline pom.xml:128]: Old workarounds that are no longer needed.

File: src/test/java/com/amazonaws/codepipeline/jenkinsplugin/AWSClientsTest.java [hunk 14]
```diff
@@ -15,7 +15,7 @@
 package com.amazonaws.codepipeline.jenkinsplugin;
 
 import static org.junit.Assert.assertEquals;
-import static org.mockito.Matchers.any;
+import static org.mockito.ArgumentMatchers.any;
```
Comment (basil) [inline src/test/java/com/amazonaws/codepipeline/jenkinsplugin/AWSClientsTest.java:18]: Upgrading to latest version of Mockito.

**Comment by egoramazon** [comment]
Thank you for contributting !

Would you mind excluding `[[maven-release-plugin] rollback the release of aws-codepipeline-0.47](https://github.com/jenkinsci/aws-codepipeline-plugin/pull/4/commits/0a91bea2d39a70361047127d30e2fcc3423ab2df)` commit from the pull request ? It doesn't exist in the mainline anymore.

File: pom.xml [hunk 15]
```diff
@@ -36,25 +38,25 @@
...
-        <id>lib</id>
-        <url>file://${project.basedir}/lib</url>
+      <id>repo.jenkins-ci.org</id>
+      <url>https://repo.jenkins-ci.org/public/</url>
     </repository>
   </repositories>
 
   <properties>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <revision>0.48</revision>
```
Comment (basil) [inline pom.xml:55]: 0.47 was a failed release, so choosing the next available version. See https://www.jenkins.io/doc/developer/publishing/releasing-manually/

**Comment by basil** [comment]
> Thank you for contributting !
> 
> Would you mind excluding `[[maven-release-plugin] rollback the release of aws-codepipeline-0.47](https://github.com/jenkinsci/aws-codepipeline-plugin/pull/4/commits/0a91bea2d39a70361047127d30e2fcc3423ab2df)` commit from the pull request ? It doesn't exist in the mainline anymore.

Done

**Comment by basil** [comment]
> It doesn't exist in the mainline anymore.

Force-pushing to the main branch is hostile to other developers

**Pull request merged**